typing/types.ml

```diff
@@ -436,6 +435,26 @@ let item_visibility = function
   | Sig_class (_, _, _, vis)
   | Sig_class_type (_, _, _, vis) -> vis
 
+let kind_abstract = Type_abstract { immediate = Unknown }
+
+let decl_is_abstract decl =
+  match decl.type_kind with
+  | Type_abstract _ -> true
+  | Type_record _ | Type_variant _ | Type_open -> false
+
+let find_unboxed_type decl =
```

Comment:
I tripped over this function a few times: it doesn't do any searching. Maybe rename to `extract_unboxed_type`?

Reply:
Gabriel suggested this name, and I think I prefer it to `extract_unboxed_type`, because to me that implies there's one to be extracted.  Here we're trying to find the inner unboxed type, if there is one.  But I'm open to other suggestions.